### PR TITLE
Skip unknown fields while decoding FieldValue proto objects.

### PR DESCRIPTION
**Pull request by rsgowman**

### CODE REVIEW

File: Firestore/core/test/firebase/firestore/remote/serializer_test.cc [hunk 1]
```diff
@@ -615,15 +620,54 @@ TEST_F(SerializerTest, BadTag) {
...
+
+  // Copy of the real one (from the nanopb generated document.pb.h), but with
+  // only boolean_value and integer_value.
+  struct google_firestore_v1beta1_Value_Fake {
+    bool boolean_value;
+    int64_t integer_value;
+  };
+
+  // Copy of the real one (from the nanopb generated document.pb.c), but with
+  // only boolean_value and integer_value. Also modified such that integer_value
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/remote/serializer_test.cc]: Entirely optional: using a named variable could make the latter part of the comment unnecessary:
```cpp
const int invalid_tag = 31;
// ...
PB_FIELD(invalid_tag, ...
```

File: Firestore/core/test/firebase/firestore/remote/serializer_test.cc [hunk 2]
```diff
@@ -615,15 +620,54 @@ TEST_F(SerializerTest, BadTag) {
...
+  const pb_field_t google_firestore_v1beta1_Value_fields_Fake[2] = {
+      PB_FIELD(1, BOOL, SINGULAR, STATIC, FIRST,
+               google_firestore_v1beta1_Value_Fake, boolean_value,
+               boolean_value, 0),
+      PB_FIELD(31, INT64, SINGULAR, STATIC, OTHER,
+               google_firestore_v1beta1_Value_Fake, integer_value,
+               boolean_value, 0),
+  };
+
+  // craft the bytes. boolean_value has a smaller tag, so it'll get encoded
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/remote/serializer_test.cc]: Ultranit: capitalize.

File: Firestore/core/test/firebase/firestore/remote/serializer_test.cc [hunk 3]
```diff
@@ -615,15 +620,54 @@ TEST_F(SerializerTest, BadTag) {
...
+  ExpectFailedStatusDuringFieldValueDecode(
+      Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
+}
+
+TEST_F(SerializerTest, BadFieldValueTagWithOtherValidTagsPresent) {
+  // A bad tag should be ignored, in which case, we should successfully
+  // deserialize the rest of the bytes as if it wasn't there. To craft these
+  // bytes, we'll use the same technique as
+  // EncodesFieldValuesWithRepeatedEntries (so go read the comments there
+  // first.)
```
[the remark this answers is not in the document]
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/remote/serializer_test.cc]: I saw it too!

File: Firestore/core/test/firebase/firestore/remote/serializer_test.cc (hunk 1 above)
Comment (rsgowman) [reply Firestore/core/test/firebase/firestore/remote/serializer_test.cc]: Done, though I ended up keeping most of the comment.

File: Firestore/core/test/firebase/firestore/remote/serializer_test.cc (hunk 2 above)
Comment (rsgowman) [reply Firestore/core/test/firebase/firestore/remote/serializer_test.cc]: done.

**Pull request merged**